server/controllers/stock/reports/stock/exit_report.js

```diff
@@ -0,0 +1,166 @@
+  const optionReport = _.extend(options, pdfOptions, {
+    filename : 'REPORT.STOCK.EXIT_REPORT',
+  });
+
+  // set up the report with report manager
+  try {
+    report = new ReportManager(STOCK_EXIT_REPORT_TEMPLATE, req.session, optionReport);
+    params = JSON.parse(req.query.params);
+  } catch (e) {
+    return next(e);
+  }
+
+  /**
+   * get into the data object :
+   * - exitToPatient : stock exit to patient data
+   * - exitToService : stock exit to service data
+   * - exitToDepot : stock exit to depot data
+   * - exitToLoss : stock exit as loss
+   */
+  try {
+    data = collect(params);
+  } catch (e) {
+    return next(e);
+  }
+
+  return fetchDepotDetails(params.depotUuid)
+    .then(depot => {
+      params.depotName = depot.text;
+      return data;
+    })
+    .then(groupCollection)
+    .then((bundle) => {
+      _.extend(bundle, params);
+
+      return report.render(bundle);
+    })
+    .then((result) => {
+      res.set(result.headers).send(result.report);
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+ * fetchDepotDetails
+ * @param {number} depotUuid depot uuid
+ */
+function fetchDepotDetails(depotUuid) {
+  const query = 'SELECT text FROM depot WHERE uuid = ?';
+  return db.one(query, [db.bid(depotUuid)]);
+}
+
+/**
+ * @function groupCollection
+ * @description group collected data by inventory
+ */
+function groupCollection(exitCollection) {
+  const collection = {};
+
+  // exit to patient
+  collection.exitToPatient = _.groupBy(exitCollection.exitToPatient, row => row.text);
+  collection.exitToPatient = _.map(collection.exitToPatient, formatExit);
+  collection.exitToPatientEmpty = (Object.keys(collection.exitToPatient).length === 0);
+
+  // exit to service
+  collection.exitToService = _.groupBy(exitCollection.exitToService, row => row.text);
+  collection.exitToService = _.map(collection.exitToService, formatExit);
+  collection.exitToServiceEmpty = (Object.keys(collection.exitToService).length === 0);
+
+  // exit to depot
+  collection.exitToDepot = _.groupBy(exitCollection.exitToDepot, row => row.text);
+  collection.exitToDepot = _.map(collection.exitToDepot, formatExit);
+  collection.exitToDepotEmpty = (Object.keys(collection.exitToDepot).length === 0);
+
+  // exit to loss
+  collection.exitToLoss = _.groupBy(exitCollection.exitToLoss, row => row.text);
+  collection.exitToLoss = _.map(collection.exitToLoss, formatExit);
+  collection.exitToLossEmpty = (Object.keys(collection.exitToLoss).length === 0);
+
+  return collection;
+}
+
+/**
+ * @function formatExit
+ */
+function formatExit(value, index) {
+  return {
+    inventory_name : index,
+    inventory_unit : value && value[0] ? value[0].unit_text : '',
+    inventory_stock_exit_data : value,
+    inventory_stock_exit_quantity : _.sumBy(value, 'quantity'),
+  };
+}
+
+/**
+ * @function collect
+ * @param {object} params query parameters
+ * @return {object} { exitToPatient: [], exitToService: [], exitToDepot: [], exitToLoss: [] }
```

Comment:
[An async function always returns a promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#Return_value).  This means that whatever calls this function should call it like this:
```js
collection(params).then().catch()
```